File: M3 Intermediate DSA/src/L9_Arrays_Sliding_Window/Assignment1.java
package L9_Arrays_Sliding_Window;

public class Assignment1 {

  public int solve(int[] A, int B, int C) {
    int N = A.length;
    int sum = 0;

    for (int index = 0; index < B; index++) {
      sum += A[index];
    }

    if (sum == C) {
      return 1;
    }

    int s = 1;
    int e = B;

    while (e < N) {
      sum = sum - A[s-1] +  A[e];

      if (sum == C) {
        return 1;
      }

      s++;
      e++;
    }

    return 0;
  }

}

File: M3_Intermediate_DSA/L7_Sliding_Window/assignment1.go
package main

func SolveA1(A []int, B int, C int) int {
	N := len(A)
	sum := 0

	for index := 0; index < B; index++ {
		sum += A[index]
	}

	if sum == C {
		return 1
	}

	s := 1
	e := B

	for e < N {
		sum -= A[s-1]
		sum += A[e]

		if sum == C {
			return 1
		}

		s++
		e++
	}

	return 0
}
